Lib/importlib/metadata/_meta.py

```diff
@@ -43,6 +45,18 @@ def json(self) -> Dict[str, Union[str, List[str]]]:
         A JSON-compatible form of the metadata.
         """
 
+    @property
+    def authors(self) -> set[Ident]:
+        """
+        Minimal parsing for "Author" and "Author-email" fields.
+        """
+
+    @property
+    def maintainers(self) -> set[Ident]:
+        """
+        Minimal parsing for "Maintainer" and "Maintainer-email" fields.
+        """
```

Comment:
When these are introduced, they will invalidate the compliance with the Protocol for any implementations except the built-in one. I suspect other libraries are allowed to implement their own `PackageMetadata`. It may be okay to introduce this change, or it may be a backward-incompatibility that needs to be considered.

Reply:
If it's not OK, what are the alternatives? I can leave the methods on the implementation but remove them from the protocol. Or I can move the new methods to `PackageMetadataV2(PackageMetadata)`, a child protocol.

Reply:
I don't know the best approach. I'm not sure how rigorous checks for protocol compliance are. Apparently this protocol isn't decorated with [runtime_checkable](https://docs.python.org/3/library/typing.html#typing.runtime_checkable), suggesting that it mainly serves as documentation and not runtime behavior.

> Or I can move the new methods to `PackageMetadataV2(PackageMetadata)`, a child protocol.

Eww. I don't like that at all. That can't possibly be the best way to evolve a protocol. I remember from C APIs that they were very struct about evolution, meaning that public APIs like the Win32 API would evolve the API once, from CallName to CallNameEx, meaning that after a year or two, the standard was for everyone to use the `Ex` calls.

As I think about it more, the `V2` concept isn't horrible. It does actually model the evolution of the protocol.

A part of me wishes there was an abstraction between the underlying versions of the protocol and the current recommended protocol, e.g. `PackageMetadataLatest` except without the `Latest` wart.

I guess that's what bothers me the most is the name is now responsible for two concerns, what the protocol is attempting to model and what iteration it is.

Here's what I recommend - let's get this change released as drafted to `importlib_metadata` and CPython 3.13 alpha and see if there are any adverse effects and learn/adapt. I may be overthinking the concern.